src/mpf.rs

```diff
@@ -119,13 +120,26 @@ impl Mpf {
         }
     }
 
-    pub fn get_str(&mut self, n_digits: i32, base: i32, exp: &mut c_long) -> String{
-        let c_str = CString::new("").unwrap();
-        let out;
-        unsafe{
-            out = CString::from_raw(__gmpf_get_str(c_str.into_raw(), exp, base, n_digits, &mut self.mpf));
+    pub fn get_str(&self, n_digits: i32, base: i32, exp: &mut c_long) -> String {
+        use std::ptr::null_mut;
+        if n_digits == 0 {
+            // Maximal significant digits requested. Let GMP compute the length
+            // and allocate the space required.
+            unsafe {
+                let p = __gmpf_get_str(null_mut(), exp, base, n_digits, &self.mpf);
+                // De-allocates the GMP string on drop.
+                GString::from_raw(p).to_str().unwrap().to_string()
+            }
+        } else {
+            // From mpf/get_str.c.
+            let bytes = unsafe {
+                let mut bytes: Vec<u8> = uninitialized();
+                let c_str: *mut c_char = transmute(bytes.as_mut_ptr());
```

Comment:
No need for a worrying `transmute`—you can just `bytes.as_mut_ptr() as *mut c_char`.